lib/pure/os.nim

```diff
@@ -2088,6 +2095,221 @@ iterator walkDir*(dir: string; relative=false): tuple[kind: PathComponent, path:
               k = getSymlinkFileKind(path)
             yield (k, y)
 
+type
+  OpenDirStatus* = enum  ## enumeration specifying status of opening a directory
+    odUnknownError, ## unrecognized error
+    odOpenOk,       ## the directory was opened OK
+    odNotDir,       ## error, not a directory: a file with the same path exists
+    odAccessDenied, ## permission denied / access denied error
+    odNotFound      ## no such path
+  WalkStepKind* = enum  ## status of the step of walking through directory
+    wsOpenDir,     ## step for attempt to open directory
+    wsEntryOk,     ## step when entry is OK
+    wsEntryBad     ## a broken symlink on posix, where it's unclear if it
+                   ## points to a file or directory
+    wsInterrupted  ## the directory handle got invalidated during reading
+  WalkStep* = object  ## step of walking through directory
+    path*: string                ## the path that this step applies to
+    code*: OSErrorCode           ## OS error code for the step
+    case kind*: WalkStepKind     ## discriminator
+    of wsOpenDir:
+      openStatus*: OpenDirStatus ## status of opening the directory
+    of wsEntryOk:
+      entryType*: PathComponent  ## path component
+    of wsEntryBad, wsInterrupted:
+      discard
+
+proc `$`*(s: WalkStep): string =
```

Comment:
this results in non-standard formatting, ideally should look like builtin `$` for obj (in particular, double quotes not single quotes for strings);
maybe all that's needed is to add `$` for OSErrorCode and then the builtin `$` would just work

Reply:
I don't know why it did not work, strange. Now, after I united openDirStatus and WalkStep enums it started to work as a bultin indeed.